cmake/Tools/Platform/Android/android_post_build.py

```diff
@@ -0,0 +1,256 @@
+        raise AndroidPostBuildError(f"Assets have not been built for this project at ({project_root}) yet. "
+                                    f"Please run the AssetProcessor for this project first.")
+
+    # Clear out the target folder first
+    safe_clear_folder(target_layout_root)
+
+    # Copy/Link the contents to the target folder
+    synchronize_folders(android_cache_folder, target_layout_root)
+
+
+def post_build_action(android_app_root: Path, native_build_folder: str, build_config: str, project_root: Path, gradle_version: Version,
+                      asset_mode: str, asset_bundle_folder: str):
+    """
+    Perform the post-build logic for android native builds that will prepare the output folders by laying out the asset files
+    to their locations before the APK is generated.
+
+    :param android_app_root:    The root path of the 'app' project within the Android Gradle build script
+    :param native_build_folder: (For Gradle versions 7.x+) Gradle uses an intermediate sub-folder The under the 'app' project.
+    :param project_root:        The root of the project that the APK is being built for
+    :param gradle_version:      The version of Gradle used to build the APK (for validation)
+    :param build_config:        The native build configuration (Debug, Profile, Release)
+    :param asset_mode:          The desired asset mode to determine the layout rules
+    :param asset_bundle_folder: (For PAK asset modes) the location of where the PAK files are expected.
+    """
+
+    android_app_root_path = Path(android_app_root)
+    if not android_app_root_path.is_dir():
+        raise AndroidPostBuildError(f"Invalid Android Gradle build path: {android_app_root_path} is not a directory or does not exist.")
+
+    if gradle_version < MINIMUM_ANDROID_GRADLE_PLUGIN_VER:
+        raise AndroidPostBuildError(f"Android gradle plugin versions below version {MINIMUM_ANDROID_GRADLE_PLUGIN_VER} is not supported.")
+    logger.info(f"Applying post-build for android gradle plugin version {gradle_version}")
+
+    # Validate the build directory exists
+    app_build_root = android_app_root_path / 'build'
+    if not app_build_root.is_dir():
+        raise AndroidPostBuildError(f"Android gradle build path: {app_build_root} is not a directory or does not exist.")
+
+    target_layout_root = android_app_root_path / 'src' / 'main' / 'assets'
+
+    if asset_mode == ASSET_MODE_LOOSE:
+
+        apply_loose_layout(project_root=project_root,
+                           target_layout_root=target_layout_root,
+                           android_app_root_path=android_app_root_path,
+                           native_build_folder=native_build_folder,
+                           build_config=build_config)
+
+    elif asset_mode == ASSET_MODE_PAK:
+
+        apply_pak_layout(project_root=project_root,
+                         target_layout_root=target_layout_root,
+                         asset_bundle_folder=asset_bundle_folder)
+
```

Comment:
should this return an error code?
```suggestion
    return 0
```

Reply:
Actually I had switched to using exceptions instead and the `__main__` handler catches `AndroidPostBuildError` so that I didnt need a bunch if error code checks and returns.